tests/Spec/Nri/Ui/Checkbox.elm

```diff
@@ -0,0 +1,131 @@
+module Spec.Nri.Ui.Checkbox exposing (..)
+
+import Accessibility.Aria as Aria
+import Accessibility.Role as Role
+import Expect
+import Html.Styled exposing (..)
+import Nri.Ui.Checkbox.V7 as Checkbox
+import ProgramTest exposing (..)
+import Spec.KeyboardHelpers as KeyboardHelpers
+import Test exposing (..)
+import Test.Html.Query as Query
+import Test.Html.Selector as Selector
+
+
+spec : Test
+spec =
+    describe "Nri.Ui.Checkbox.V7"
+        [ describe "'checkbox' role" hasCorrectRole
+        , describe "aria-checked" hasAriaChecked
+        ]
+
+
+hasCorrectRole : List Test
+hasCorrectRole =
+    [ test "has role checkbox" <|
+        \() ->
+            program
+                |> ensureHasRoleCheckbox
+                |> done
+    ]
+
+
+hasAriaChecked : List Test
+hasAriaChecked =
+    [ test "aria-checked reflects the state of the checkbox" <|
+        \() ->
+            program
+                |> ensureAriaCheckedIsFalse
+                |> pressSpace
+                |> ensureAriaCheckedIsTrue
+                |> pressSpace
+                |> ensureAriaCheckedIsFalse
+                |> done
+    ]
+
+
+ensureHasRoleCheckbox : TestContext -> TestContext
+ensureHasRoleCheckbox context =
+    context
+        |> ensureView
+            (Query.findAll [ Selector.attribute Role.checkBox ]
+                >> Query.count (Expect.equal 1)
+            )
```

Comment:
```suggestion
        |> ensureViewHas [ Selector.attribute Role.checkBox ]
```

I think for this helper and the 2 helpers below it, `ensureViewHas` can make the assertion a bit more compact and remove the need for `Query`. (At which point, I wonder if the helpers should actually be inlined, instead of being separate functions?)